packages/cart/src/models/shipping-method-adjustment.ts

```diff
@@ -1,15 +1,25 @@
 import { generateEntityId } from "@medusajs/utils"
-import { BeforeCreate, Entity, ManyToOne, OnInit } from "@mikro-orm/core"
+import {
+  BeforeCreate,
+  Entity,
+  ManyToOne,
+  OnInit,
+  Property,
+} from "@mikro-orm/core"
 import AdjustmentLine from "./adjustment-line"
 import ShippingMethod from "./shipping-method"
 
 @Entity({ tableName: "cart_shipping_method_adjustment_line" })
-export default class ShippingMethodAdjustmentLine extends AdjustmentLine {
+export default class ShippingMethodAdjustment extends AdjustmentLine {
   @ManyToOne(() => ShippingMethod, {
-    joinColumn: "shipping_method",
-    fieldName: "shipping_method_id",
+    onDelete: "cascade",
+    nullable: true,
+    index: "IDX_adjustment_shipping_method_id",
   })
-  shipping_method: ShippingMethod
+  shipping_method: ShippingMethod | null
+
+  @Property({ columnType: "text" })
+  shipping_method_id: string
```

Comment:
```suggestion
  @Property({ columnType: "text", nullable: true })
  shipping_method_id: string | null
```

should we do this here as well + Optional fields?